fan-xiaomi.js

```diff
@@ -367,13 +435,23 @@ Natural
             activeElement.classList.remove('active')
         }
         let direct_speed_int = Number(direct_speed)
-        if (direct_speed_int <= 20) {
+        
+        if (model === 'dmaker.fan.p5') { //p5 does not report direct_speed and natural_speed
+            direct_speed_int = speed_list[speed]
+            if (mode === 'nature') {
+                natural_speed = true
+            } else if (mode === 'normal') {
+                natural_speed = false
+            }
+        }
+        
+        if (direct_speed_int <= speed_list['Level 1']) {
```

Comment:
```suggestion
        if (direct_speed_int <= speed_list[0]) {
```
Using arrays for speed_list